authorization-server/oauth2.js

```diff
@@ -129,7 +129,10 @@ server.exchange(oauth2orize.exchange.clientCredentials((client, scope, done) =>
  */
 server.exchange(oauth2orize.exchange.refreshToken((client, refreshToken, scope, done) => {
   db.refreshTokens.find(refreshToken)
-  .then(foundRefreshToken => validate.refreshToken(foundRefreshToken, refreshToken, client))
+  .then(foundRefreshToken => {
+    validate.refreshToken(foundRefreshToken, refreshToken, client)
+    return foundRefreshToken;
```

Comment:
Huh. I think you're right here. However, it's been a long time since I've looked at this code. Sad, I know. ;-)

Well, thank you for making this pull request. I think you meant to add a semicolon on line 133. (also my fault for not hooking up the linter to travis CI for PR's) Do you mind writing a unit test that fails at first but then this fixes it?

For unit test it would be here:
[validate.js#L181](https://github.com/FrankHassanabad/Oauth2orizeRecipes/blob/master/authorization-server/test/validate.js#L181)

Also, do you think the issue is here or is the issue within `validate.refreshToken`? I think the fix would be better in `validate.refreshToken` since its job is to return the correct token after validating it and this is the only spot its used.

Reply:
Sure, I will make a new commit with unit testing. 

You are right, ```validate.refreshToken``` is more appropriate way.